### PR TITLE
Feature/AVO-3180 select education level modal

### PR DESCRIPTION
**Pull request by ian-emsens-shd**
Screenshot 2024-05-06 at 15 00 21
Screenshot 2024-05-06 at 15 00 46

### CODE REVIEW

**Comment by ian-emsens-shd** [comment]
Translations extracted in https://github.com/viaacode/avo2-client/pull/1984/commits/d0a8ddf1609a9112f9548033b8fc273cedacc670 to avoid double merge commit

File: src/assignment/views/AssignmentCreate.tsx [hunk 1]
```diff
@@ -1,6 +1,10 @@
+import './AssignmentCreate.scss';
+import './AssignmentPage.scss';
+
 import { yupResolver } from '@hookform/resolvers/yup';
 import { Button, Container, Icon, IconName, Spacer } from '@viaa/avo2-components';
 import { type Avo } from '@viaa/avo2-types';
+import { type LomFieldSchema } from '@viaa/avo2-types/types/lom';
```
Comment (bertyhell) [inline src/assignment/views/AssignmentCreate.tsx:7]: don't do deep imports from libraries
use:
```suggestion
import { type Avo } from '@viaa/avo2-types';
```

and then
```
Avo.Lom.LomField
```

so there is a clear "public interface" of the library, and the library can change folder structure without breaking depedent sites

File: src/assignment/views/AssignmentCreate.tsx [hunk 2]
```diff
@@ -147,12 +157,22 @@ const AssignmentCreate: FunctionComponent<DefaultSecureRouteProps> = ({
 		resetForm();
 	}, [resetForm, setAssignment, defaultValues]);
 
+	const selectLevel = useCallback(
+		(lom: LomFieldSchema) => {
+			if (!assignment) return;
+			setSelectEducationLevelModalOpen(false);
+			(assignment as any).education_level_id = lom.id; // TODO
```
Comment (bertyhell) [inline src/assignment/views/AssignmentCreate.tsx:164]: TODO this PR or TODO future PR?

File: src/assignment/views/AssignmentCreate.tsx [hunk 3]
```diff
@@ -339,46 +359,64 @@ const AssignmentCreate: FunctionComponent<DefaultSecureRouteProps> = ({
 		}
 	}, [assignment, loadingInfo, setLoadingInfo]);
 
+	useEffect(() => {
+		if (!assignment || (assignment as any).education_level_id) return;
+		isUserDoubleTeacher(commonUser) && setSelectEducationLevelModalOpen(true);
```
Comment (bertyhell) [inline src/assignment/views/AssignmentCreate.tsx:364]: optional: isUserDoubleEducationLevelTeacher

File: src/assignment/views/AssignmentEdit.tsx [hunk 4]
```diff
@@ -188,6 +195,11 @@ const AssignmentEdit: FunctionComponent<AssignmentEditProps & UserProps> = ({
 		fetchContributors();
 	}, [fetchContributors]);
 
+	useEffect(() => {
+		if (!assignment || (assignment as any).education_level_id) return;
```
Comment (bertyhell) [inline src/assignment/views/AssignmentEdit.tsx:199]:
```suggestion
		if (!assignment || (assignment as any).education_level_id) return; // TODO
```

File: src/assignment/views/AssignmentOverview.tsx [hunk 5]
```diff
@@ -1,3 +1,5 @@
+import './AssignmentOverview.scss';
```
Comment (bertyhell) [inline src/assignment/views/AssignmentOverview.tsx:1]: We should get the eslint import ordering to work for scss files, so scss files are automatically sorted to the top

File: src/shared/components/SelectEducationLevelModal/SelectEducationLevelModal.tsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+	ButtonToolbar,
+	Modal,
+	ModalBody,
+	type ModalProps,
+	Select,
+	Toolbar,
+	ToolbarItem,
+	ToolbarRight,
+} from '@viaa/avo2-components';
+import { type LomFieldSchema } from '@viaa/avo2-types/types/lom';
```
Comment (bertyhell) [inline src/shared/components/SelectEducationLevelModal/SelectEducationLevelModal.tsx:14]: no deep imports from libraries, see above

File: src/shared/components/SelectEducationLevelModal/SelectEducationLevelModal.tsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+	const { user, commonUser, onConfirm, ...modal } = props;
+
+	const [educationLevels] = useLomEducationLevels();
+	const [selected, setSelected] = useState<LomFieldSchema | undefined>(undefined);
+
+	const options = useMemo(
+		() =>
+			[EducationLevelId.lagerOnderwijs, EducationLevelId.secundairOnderwijs]
+				.map((level) => educationLevels.find((item) => item.id === level))
+				.map(
```
Comment (bertyhell) [inline src/shared/components/SelectEducationLevelModal/SelectEducationLevelModal.tsx:39]: optional:
```suggestion
			educationLevels.filter(level => {
				return [EducationLevelId.lagerOnderwijs, EducationLevelId.secundairOnderwijs].includes(level.id)
				})
				.map(
```

File: src/shared/components/SelectEducationLevelModal/SelectEducationLevelModal.tsx [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+		onConfirm?: (lom: LomFieldSchema) => void;
+	};
+
+// Component
+
+const SelectEducationLevelModal: FunctionComponent<SelectEducationLevelModalProps> = (props) => {
+	const { user, commonUser, onConfirm, ...modal } = props;
+
+	const [educationLevels] = useLomEducationLevels();
+	const [selected, setSelected] = useState<LomFieldSchema | undefined>(undefined);
```
Comment (bertyhell) [inline src/shared/components/SelectEducationLevelModal/SelectEducationLevelModal.tsx:33]: PERSONAL PREFERENCE: If it is something you set intentianally to empty, i usually use null. And i leave undefined for things that are empty but were not set by me
```suggestion
	const [selected, setSelected] = useState<LomFieldSchema | null>(null);
```

File: src/shared/helpers/is-user.ts [hunk 9]
```diff
@@ -10,6 +12,20 @@ export function isUserElementaryPupil(user: CommonUserSchema) {
 	return isUser(user, SpecialUserGroup.PupilElementary);
 }
 
+export function isUserSecondaryTeacher(user: CommonUserSchema) {
+	return isUser(user, SpecialUserGroup.TeacherSecondary);
+}
+
+export function isUserDoubleTeacher(user: CommonUserSchema) {
```
Comment (bertyhell) [inline src/shared/helpers/is-user.ts:19]: add some jsdoc what a double teacher is for someone who isn't familiar
i'm guessing teacher with secundairy and elementary education levels linked?

or maybe rename to:
```suggestion
export function isUserDoubleEducationLevelTeacher(user: CommonUserSchema) {
```

**Review by bertyhell — APPROVED** [review]

File: src/assignment/views/AssignmentCreate.tsx (hunk 2 above)
Comment (ian-emsens-shd) [reply src/assignment/views/AssignmentCreate.tsx:164]: TODO future in bulk

File: src/assignment/views/AssignmentOverview.tsx (hunk 5 above)
Comment (ian-emsens-shd) [reply src/assignment/views/AssignmentOverview.tsx:1]: With a quick search it doesn't look like it's out-of-the-box due to some import order nuance

File: src/shared/components/SelectEducationLevelModal/SelectEducationLevelModal.tsx (hunk 8 above)
Comment (ian-emsens-shd) [reply src/shared/components/SelectEducationLevelModal/SelectEducationLevelModal.tsx:33]: Output of `educationLevels.find` can be `undefined` hence this state.

**Pull request merged**